sdks/python/apache_beam/runners/worker/sdk_worker.py

```diff
@@ -296,6 +319,51 @@ def initial_source_split(self, request, unused_instruction_id=None):
     return response
 
   def create_execution_tree(self, descriptor):
+    if descriptor.primitive_transform:
+      return self.create_execution_tree_from_fn_api(descriptor)
+    else:
+      return self.create_execution_tree_from_runner_api(descriptor)
+
+  def create_execution_tree_from_runner_api(self, descriptor):
+    # TODO(robertwb): Figure out the correct prefix to use for output counters
+    # from StateSampler.
+    counter_factory = counters.CounterFactory()
+    state_sampler = statesampler.StateSampler(
+        'fnapi-step%s-' % descriptor.id, counter_factory)
+
+    transform_factory = BeamTransformFactory(
+        descriptor, self.data_channel_factory, counter_factory, state_sampler,
+        self.state_handler)
+
+    pcoll_consumers = collections.defaultdict(list)
+    for transform_id, transform_proto in descriptor.transforms.items():
+      for pcoll_id in transform_proto.inputs.values():
+        pcoll_consumers[pcoll_id].append(transform_id)
+
+    @memoize
+    def get_operation(transform_id):
```

Comment:
The memoization is local to the execution of this bundle. Eventually we'll want to cache and re-used these (though not concurrently).